process/structure.go

```diff
@@ -0,0 +1,37 @@
+package process
+
+import (
+	"encoding/json"
+	"io"
+	"strings"
+	"time"
+)
+
+// Structure outputs messages written to it in a structured format, including metadata.
+type Structure struct {
+	w        io.Writer
+	metadata map[string]string
+	encoder  *json.Encoder
+}
+
+func NewStructure(w io.Writer, metadata map[string]string) *Structure {
+	var metadata_copy = map[string]string{}
+	for k, v := range metadata {
+		metadata_copy[k] = v
+	}
```

Comment:
why copy the map here? if it's necessary for some reason, this can be replaced with:
```suggestion
	metadataCopy := maps.Clone(metadata)
```
but i suspect it can be left out

Reply:
You know, I might just try to adapt or wrap the existing Logger + JSONWriter code into the job logger stuff. It's basically the achieving the same end result.

Reply:
yeah, i deleted the comment because i realised that i'd misread the benchmark and that the thing that i had said was quicker was actually slower, and i'd just misread the benchmark ahaha.

regardless, i think either approach is is totally fine, we can probably spare 1000 nanoseconds here or there